Library/Homebrew/cmd/upgrade.rb

```diff
@@ -213,37 +213,42 @@ def upgradable_dependents(kegs, formulae)
     formulae_to_upgrade = Set.new
     formulae_pinned = Set.new
 
-    formulae.each do |formula|
+    formulae_to_check = formulae
+    checked_formulae = Set.new
+
+    until formulae_to_check.empty?
```

Comment:
I worry about this still creating an infinite loop. Is this avoidable? Perhaps a max depth could be configured?

Reply:
I think that's what the 

```
next if checked_formulae.include?(formula)
```

below is for.

Reply:
@reitermarkus is correct, the trick lies in that line.